blocks/library/latest-comments/index.js

```diff
@@ -0,0 +1,209 @@
+			this.changeCommentsToShow = this.changeCommentsToShow.bind( this );
+
+			const { commentsToShow } = this.props.attributes;
+
+			this.state = {
+				latestComments: [],
+			};
+
+			this.latestCommentsRequest = getLatestComments( commentsToShow );
+
+			this.latestCommentsRequest
+				.then( latestComments => this.setState( { latestComments } ) );
+		}
+
+		componentWillReceiveProps( nextProps ) {
+			const { commentsToShow: commentToShowCurrent } = this.props.attributes;
+			const { commentsToShow: commentsToShowNext } = nextProps.attributes;
+			const { setAttributes } = this.props;
+
+			if ( commentToShowCurrent === commentsToShowNext ) {
+				return;
+			}
+
+			if ( commentsToShowNext >= MIN_COMMENTS && commentsToShowNext <= MAX_COMMENTS ) {
+				this.latestCommentsRequest = getLatestComments( commentsToShowNext );
+
+				this.latestCommentsRequest
+					.then( latestComments => this.setState( { latestComments } ) );
+
+				setAttributes( { commentsToShow: commentsToShowNext } );
+			}
+		}
+
+		toggleDisplayAvatar() {
+			const { displayAvatar } = this.props.attributes;
+			const { setAttributes } = this.props;
+
+			setAttributes( { displayAvatar: ! displayAvatar } );
+		}
+
+		toggleDisplayExcerpt() {
+			const { displayExcerpt } = this.props.attributes;
+			const { setAttributes } = this.props;
+
+			setAttributes( { displayExcerpt: ! displayExcerpt } );
+		}
+
+		toggleDisplayTimestamp() {
+			const { displayTimestamp } = this.props.attributes;
+			const { setAttributes } = this.props;
+
+			setAttributes( { displayTimestamp: ! displayTimestamp } );
+		}
+
+		changeCommentsToShow( commentsToShow ) {
+			const { setAttributes } = this.props;
+
+			setAttributes( { commentsToShow: parseInt( commentsToShow, 10 ) || 0 } );
+		}
+
+		render() {
+			const { latestComments } = this.state;
+			const { setAttributes } = this.props;
+
+			if ( ! latestComments.length ) {
+				return (
+					<Placeholder
+						icon="admin-post"
+						label={ __( 'Latest Comments' ) }
+					>
+						<Spinner />
+					</Placeholder>
+				);
+			}
+
+			const { focus } = this.props;
+			const { align, displayAvatar, displayTimestamp, displayExcerpt } = this.props.attributes;
+
+			return [
+				focus && (
+					<BlockControls key="controls">
+						<BlockAlignmentToolbar
+							value={ align }
+							onChange={ ( nextAlign ) => {
+								setAttributes( { align: nextAlign } );
+							} }
+							controls={ [ 'left', 'center', 'right', 'wide', 'full' ] }
+						/>
+					</BlockControls>
+				),
+				focus && (
+					<InspectorControls key="inspector">
+						<BlockDescription>
+							<p>{ __( 'Shows a list of your site\'s most recent comments.' ) }</p>
+						</BlockDescription>
+						<h3>{ __( 'Latest Comments Settings' ) }</h3>
+
+						<ToggleControl
+							label={ __( 'Display avatar' ) }
+							checked={ displayAvatar }
+							onChange={ this.toggleDisplayAvatar }
+						/>
+
+						<ToggleControl
+							label={ __( 'Display timestamp' ) }
```

Comment:
I think it should be “Display publish date” both here and on the Latest Posts block.